console/libretime_console/setup.py

```diff
@@ -0,0 +1,200 @@
+
+    config["rabbitmq"]["host"] = click.prompt(
+        text="Your message queue host",
+        default=config["rabbitmq"]["host"],
+        type=str,
+    )
+    config["rabbitmq"]["port"] = click.prompt(
+        text="Your message queue port",
+        default=config["rabbitmq"]["port"],
+        type=int,
+    )
+    config["rabbitmq"]["vhost"] = click.prompt(
+        text="Your message queue vhost",
+        default=config["rabbitmq"]["vhost"],
+        type=str,
+    )
+    config["rabbitmq"]["user"] = click.prompt(
+        text="Your message queue user",
+        default=config["rabbitmq"]["user"],
+        type=str,
+    )
+    config["rabbitmq"]["password"] = click.prompt(
+        text="Your message queue user password",
+        default=config["rabbitmq"]["password"],
+        type=str,
+    )
+
+    section_title("# Storage settings")
+
+    config["storage"]["path"] = click.prompt(
+        text="Your storage directory path",
+        default=config["storage"]["path"],
+        type=str,
+    )
+
+    click.echo()
+    click.confirm(click.style("Please confirm your choices", fg="red"), abort=True)
+
+    print(dump(config, indent=2), encoding="utf-8", file=sys.stdout)
+
+
+DEFAULT_MIGRATIONS_PATH = Path("./legacy/build/sql")
+DEFAULT_STORAGE_PATH = Path("/srv/libretime/data")
+
+# pylint: disable=invalid-name
+@setup.command()
+@click.pass_obj
+@click.argument("public_url")
+@click.option(
+    "--migrations-path",
+    type=click.Path(path_type=Path),
+    default=DEFAULT_MIGRATIONS_PATH,
+)
+@click.option(
+    "--storage-path",
+    type=click.Path(path_type=Path),
+    default=DEFAULT_STORAGE_PATH,
+)
+@click.option(
+    "--icecast-admin-password",
+    type=str,
+)
+def db(
+    obj: Dict[str, Any],
+    public_url: str,
+    migrations_path: Path,
+    storage_path: Path,
+    icecast_admin_password: str,
+):
+    """
+    Setup the database.
+    """
+    config = Config(filepath=obj["config_filepath"])
+
+    # get_setting(config, "schema")
+
+    with get_db_session(config) as session:
+        for migration_filename in (
+            "schema.sql",
+            "defaultdata.sql",
+        ):
+            migration_filepath = migrations_path / migration_filename
+            sql = migration_filepath.read_text(encoding="utf-8")
+            session.execute(sql)
+
+    set_storage(config, storage_path)
+
+    set_setting(config, "allowed_cors_urls", public_url.strip())
+
+    set_setting(config, "default_icecast_password", icecast_admin_password)
+
+    with get_db_session(config) as session:
+        for keys in product(["s1", "s2", "s3", "s4"], ["pass", "admin_pass"]):
+            session.execute(
+                """
+                UPDATE cc_stream_setting
+                SET value = %(value)s
+                WHERE keyname = %(key)s
+                """,
+                {
+                    "key": "_".join(keys),
+                    "value": icecast_admin_password,
+                },
+            )
```

Comment:
This is the equivalent code the is run the web installer. I am currently working on moving this code to a django command